crates/yamlpatch/src/lib.rs

```diff
@@ -244,10 +273,8 @@ fn apply_single_patch(
 
             // Finally, put our patch back into the overall content.
             let mut patched_content = content.to_string();
-            patched_content.replace_range(
-                feature.location.byte_span.0..feature.location.byte_span.1,
-                &patched_feature,
-            );
+            let replacement_range = compute_replacement_range(&feature, document);
+            patched_content.replace_range(replacement_range, &patched_feature);
```

Comment:
Not a blocker, just thinking out loud: it's not super ideal that we need these helpers now, since I could easily see us forgetting to call them in relevant places in the future. We might want to do more type state encoding here, i.e. have a `RawSpan` and `AdjustedSpan` or whatever and only allow replacements on ranges with the latter.

(That isn't necessarily the right approach either, just something that might help.)